File: 3.2/mobile/lab3/lab3/project1/app/src/androidTest/java/com/example/project1/MainActivityTest.java
package com.example.project1;

import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.ext.junit.rules.ActivityScenarioRule;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.*;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.*;

@RunWith(AndroidJUnit4.class)
public class MainActivityTest {
    @Rule
    public ActivityScenarioRule<MainActivity> activityRule = new ActivityScenarioRule<>(MainActivity.class);

    @Test
    public void testInitialTextDisplayed() {
        onView(withId(R.id.tvInfo))
                .check(matches(withText(R.string.try_to_guess)));
    }

    @Test
    public void testUserGuess() {
        onView(withId(R.id.etInput)).perform(typeText("50"), closeSoftKeyboard());
        onView(withId(R.id.bControl)).perform(click());

        // Проверяем, что TextView с информацией изменил своё состояние (текст может быть "Загаданное число больше" или "Загаданное число меньше")
        onView(withId(R.id.tvInfo)).check(matches(isDisplayed()));
    }
}

File: 3.2/mobile/lab3/lab3/Calculator16/app/src/androidTest/java/com/example/calculator16/MainActivityTest.kt
package com.example.calculator16

import androidx.test.core.app.ActivityScenario
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.*
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.LargeTest
import org.hamcrest.CoreMatchers.containsString
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
@LargeTest
class MainActivityTest {

    @Test
    fun testInitialDisplay() {
        ActivityScenario.launch(MainActivity::class.java)
        onView(withId(R.id.tvDisplay)).check(matches(withText("0")))
    }

    @Test
    fun testNumberInput() {
        ActivityScenario.launch(MainActivity::class.java)
        onView(withId(R.id.btn1)).perform(click())
        onView(withId(R.id.btn2)).perform(click())
        onView(withId(R.id.tvDisplay)).check(matches(withText(containsString("12"))))
    }

    @Test
    fun testConversionWithoutInputShowsError() {
        ActivityScenario.launch(MainActivity::class.java)
        onView(withId(R.id.btnClear)).perform(click())
        onView(withId(R.id.btnConvertUK)).perform(click())
        onView(withText(R.string.error_no_input)).check(matches(isDisplayed()))
    }

    @Test
    fun testNavigationToInfoFragment() {
        ActivityScenario.launch(MainActivity::class.java)
        onView(withId(R.id.btnInfo)).perform(click())
        onView(withText(R.string.info_message)).check(matches(isDisplayed()))
    }
}
